client/src/controllers/DropdownController.ts

```diff
@@ -307,6 +309,6 @@ export class DropdownController extends Controller<HTMLElement> {
   }
 
   disconnect() {
-    this.tippy?.destroy();
+    this.tippy?.hide();
```

Comment:
I'm not sure this is the right solution. Keeping the Tippy instance indefinitely after the Stimulus controller is disconnected can cause memory leak issues. We have places where the controller is replaced multiple times, e.g. in the concurrent editing notifications.

Maybe the solution is to somehow make sure that the necessary options/elements to instantiate Tippy is retained after the initial render and after the drag-and-drop. Not sure if it does, but the `keepMounted` value may help.

Reply:
Some features like the concurrent editing notifications (see `SessionController`) works by swapping the HTML with a freshly-rendered fragment from the server. If the fragment contains an element with `data-controller="w-dropdown"`, I doubt `this` will be the same object as the one from the previous fragment. For example, in the `SessionController`, a dropdown might only be used if there are sessions that have made changes, or if there are many other active sessions. When a swap occurs, it may or may not even contain a dropdown.

In short, there are cases where it may not be the same controller instance that gets unmounted and remounted. If the element is unmounted and removed from the DOM, or just unmounted indefinitely, the tippy instance will only be destroyed by the garbage collector. Not to mention that we'd assume it can be garbage collected (no other leftover references to the tippy), which I can't be sure without doing some profiling.

Reply:
Thank you for explaining !

Reply:
@laymonage `keepMouted` is working but there is obvious issue with this, like if someone had to go through big list to change item's order so when item is dragged and on every item controller connects and disconnects which appends tippy element,

E.g,

https://github.com/user-attachments/assets/dada732e-b0d1-431f-bba2-f148a52b0447

I'm thinking for two solutions for this based on use case

1. if it is not necessary to append tippy element when `keepMounted=true` then instead append we can replace new target.
2. Or if it's necessary append instead then we can create new value data attribute (keep-mounted-is-append; default : true)  if false then not to append but replace.

https://github.com/wagtail/wagtail/blob/61b50d3c28fb55f8c56bae78bd6f7c44e866e022/client/src/controllers/DropdownController.ts#L183